cadl-tests/src/test/java/com/dictionary/UnknownValueClientTest.java

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.dictionary;
+
+import com.azure.core.util.BinaryData;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.Map;
+
+class UnknownValueClientTest {
+
+    UnknownValueClient client = new UnknownValueClientBuilder().buildClient();
+
+    @Test
+    void get() {
+       Map<String, Object> response = client.get();
+       Assertions.assertEquals(1, response.get("k1"));
+       Assertions.assertEquals("hello", response.get("k2"));
+       Assertions.assertEquals(null, response.get("k3"));
+    }
+
+    @Test
+    void put() {
+        client.putWithResponse(BinaryData.fromString("{\"k1\":1,\"k2\":\"hello\",\"k3\":null}"), null);
```

Comment:
there is no other way?

Reply:
I tried to use `JsonNodeFactory` to build the map, but the problem is that we can't set null value to `ObjectNode`
![image](https://user-images.githubusercontent.com/87355844/197992454-7fa759fa-6288-48c6-bb87-ee4d4cc98dca.png)

Reply:
Please do some investigate yourself https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/node/NullNode.html

Reply:
I understand this is doable, I can use `map.put("k3", NullNode.instance);`. But this [`put`](https://fasterxml.github.io/jackson-databind/javadoc/2.7/com/fasterxml/jackson/databind/node/ObjectNode.html#put(java.lang.String,%20com.fasterxml.jackson.databind.JsonNode)) signature is deprecated. So I did not continue to try this way.

Reply:
please just read it


@Deprecated
public JsonNode put(String fieldName,
                      JsonNode value)
Deprecated. Since 2.4 use either set(String,JsonNode) or replace(String,JsonNode),
Method that will set specified field, replacing old value, if any.

Reply:
Thanks for pointing out, sorry that I did not download the source but just look at `.class` file, so did not notice this description. 

I can use below way to test the `put` method, but from the test perspective, do you think `BinaryData.fromObject()` bring more benefits than `BinaryData.fromString()`? We now understand we can pass in `ObjectNode` to `BinaryData.fromObject()`, and null value will be serialized. But for test, personally I think either way works well. Using `BinaryData.fromString()`, we can easily build the `BinaryData` through the expected response in the log.

![image](https://user-images.githubusercontent.com/87355844/198193236-b6af9796-3d82-44b2-b4ae-5014d7f73c22.png)

Reply:
No from test, yes from knowledge. Please change it.